tests/util/verify.sh

```diff
@@ -169,10 +177,124 @@ _verify_like() {
                         if [[ "$comm" =~ ^([a-zA-Z0-9_]+-)+ ]]; then
                             break
                         fi
-                        _err_exit "$msg" "$out"
+                        return 1
                     fi
                 done
             done
         done
     fi
+
+    return 0
+}
+
+
+# Verify that $out is the same as $expected.  If they are not the same,
+# exponentially back off and try again.
+__verify_with_retry() {
+    local cmp_func=$1
+    local cmd=$2
+    local expected=$3
+    local max_attempts=6
+    local attempt=0
+
+    while true; do
+      out=$($cmd 2>&1)
+
+      if $cmp_func "$out" "$expected"; then
+          return
+      fi
+
+      if (( $attempt >= $max_attempts )); then
+          __err_exit "$cmd" "$out"
```

Comment:
Don't we need $msg here as well to indicate which snip in the test is failing. Sometimes different snips in the same doc use the same command more than once. It also seems that without the the snip function name in the message, it will be harder, in general, to track back to where in the test the failing call is coming from.

Reply:
Most of the $msg arguments that I saw were just the name of the function anyway.  Just to be clear, this is the name of the function from the snips.sh file.  For example, here's what it prints from that "No resources found" failure that was discussed in Slack:
```
VERIFY FAILED snip_set_up_the_cluster_5: No resources found
```
It's probably confusing since it uses the argument name "$cmd" instead of "$func".  I just wanted to separate it from the comparison function argument.  I'll look at finding a better name, since "command" is overloaded.

Does that make sense?  Let me know if I misunderstood what you suggested.

Reply:
Ah, that makes perfect sense. I didn't look closely enough and I thought it was going to display something like `VERIFY FAILED "curl bla bla bla": No resources found`.